homeassistant/components/homewizard/coordinator.py

```diff
@@ -28,15 +28,18 @@ def __init__(
         """Initialize Update Coordinator."""
 
         self.host = host
-        super().__init__(hass, _LOGGER, name="", update_interval=UPDATE_INTERVAL)
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=UPDATE_INTERVAL)
 
     async def _async_update_data(self) -> dict:
         """Fetch all device and sensor data from api."""
 
         async with async_timeout.timeout(10):
 
             if self.api is None:
-                self.api = await self._initialize_api()
+                await self.initialize_api()
+
+            # Tell MyPi that self.api is set (silence attr-defined)
+            assert self.api is not None
```

Comment:
You can retrun the api object from `self.initialize_api()` so Mypy would know without having to use `assert` (which you shouldn't do in production code).

Also please either fully type functions or classes or don't annotate at all (eg. `__init__()` is missing an annotation for `host`) and try to specify generic types like `dict` (in this case eg. `_async_update_data(self) -> dict[str, Any]`)

You may also don't have to use an optional for self.api and self.host - just annotate it at class level `api: aiohwenergy.HomeWizardEnergy` or `host: str` without setting it to None and make sure you don't access it before you assigned something - eg. in `__init__()`. On the other hand - if you assign in `__init__` you can just omit the class level assignment.

Reply:
Sorry for the delay, but thanks!
I've fixed some issues I had with initialising the API while having the required behaviour. I think it should be fixed now.

I also annotated everything I can think of and passed `mypy`, so I've added it to `.strict-typing`.